components/sync15/src/telemetry.rs

```diff
@@ -254,6 +254,8 @@ pub enum SyncFailure {
 
 pub fn sync_failure_from_error(e: &Error) -> SyncFailure {
     SyncFailure::Unexpected {
+        // TODO: Distinguish between error types, truncate, and anonymize
```

Comment:
We should have a followup to do this - hopefully the SyncStatus/SyncResult stuff we landed recently makes that a little easier.